hedera-node/src/main/java/com/hedera/services/stream/RecordStreamObject.java

```diff
@@ -38,62 +38,71 @@
 import java.time.Instant;
 
 /**
- * Contains a ExpirableTxnRecord, its related Transaction, and consensus Timestamp of the Transaction.
+ * Contains a TransactionRecord, its related Transaction, and consensus Timestamp of the Transaction.
  * Is used for record streaming
  */
 public class RecordStreamObject extends AbstractSerializableHashable implements Timestamped,
 		SerializableRunningHashable {
 	private static final long CLASS_ID = 0xe370929ba5429d8bL;
-	public static final int CLASS_VERSION = 1;
+	static final int CLASS_VERSION = 1;
 
-	//TODO: confirm the max length;
 	private static final int MAX_RECORD_LENGTH = 64 * 1024;
 	private static final int MAX_TRANSACTION_LENGTH = 64 * 1024;
 
-	/** The fast-copyable equivalent of the gRPC transaction record for the record stream file */
-	private ExpirableTxnRecord expirableTxnRecord;
-
-	/** The gRPC transaction for the record stream file */
+	/* The gRPC transaction for the record stream file */
 	private Transaction transaction;
+	private TransactionRecord transactionRecord;
+	/* The fast-copyable equivalent of the gRPC transaction record for the record stream file */
+	private ExpirableTxnRecord fcTransactionRecord;
 
-	/**
-	 * the consensus timestamp of this {@link ExpirableTxnRecord} object,
-	 * this field is used for deciding whether to start a new record stream file,
-	 * and for generating file name when starting to write a new record stream file;
-	 * this field is not written to record stream file
-	 */
+	/* The consensus timestamp of this object's transaction; determines when to start a
+	 * new record stream file, and the name to use for a new file if started. However,
+	 * this field is NOT itself included in the record stream. */
 	private Instant consensusTimestamp;
 
-	/**
-	 * this RunningHash instance encapsulates a Hash object which denotes a running Hash calculated from
-	 * all RecordStreamObject in history up to this RecordStreamObject instance
-	 */
+	/* The running hash of all objects streamed up to and including this consensus time. */
 	private RunningHash runningHash;
 
 	public RecordStreamObject() {
 	}
 
-	public RecordStreamObject(final ExpirableTxnRecord expirableTxnRecord,
-			final Transaction transaction, final Instant consensusTimestamp) {
-		// configurable 50/100/150 bytes
-		this.expirableTxnRecord = expirableTxnRecord;
+	public RecordStreamObject(
+			final TransactionRecord transactionRecord,
```

Comment:
Do we need `RecordStreamObject` to have grpc type as well?

Reply:
There are two constructors, this one is only used by the unit test.

For an instance constructed w/o an explicit `TransactionRecord`, the `ensureNonNullGrpcRecord()` record is called to translate from the `ExpirableTxnRecord`.

The goal here is to avoid calling `ExpirableTxnRecord.asGrpc()` in the `handleTransaction` thread.

Reply:
I see .. Makes sense !